src/statusTile.test.js

```diff
@@ -0,0 +1,36 @@
+// @flow
+const { createStatusTile, updateStatusTile } = require('./statusTile');
+
+test('it creates a div with a proper class name, a "Prettier" text node and a tooltip', () => {
+  atom = {
+    config: { get: jest.fn(() => true) },
+    tooltips: { add: () => ({ prop: '...' }) },
+  };
+
+  const classes = [];
+  global.document = {
+    createElement: jest.fn(() => ({
+      classList: {
+        add: jest.fn(arg => classes.push(arg)),
+      },
+      dataset: {},
+      appendChild: jest.fn(),
+    })),
+    createTextNode: jest.fn(arg => arg),
+  };
+
+  const div = createStatusTile();
+  expect(div).toBeDefined();
+  expect(div.dataset.formatOnSave).toBe('enabled');
+  expect(classes.includes('prettier-atom-status-tile')).toBe(true);
+  expect(document.createTextNode).toHaveBeenCalledWith('Prettier');
+
+  const disposable = { add: jest.fn() };
+  const tooltip = updateStatusTile(disposable, div);
+  expect(tooltip.prop).toBe('...');
+  expect(disposable.add).toHaveBeenCalled();
+
+  tooltip.dispose = jest.fn();
+  updateStatusTile(disposable, div);
+  expect(tooltip.dispose).toHaveBeenCalled();
+});
```

Comment:
It seems we are testing multiple different functions in the same test? I'm not sure if this would work, but can we split them so that it's a bit easier to see what each one does, and also so that each function is tested in isolation of the other one? Something like this, although it requires creating and mocking out a `disposeCurrentTooltip` function or something:

```js
beforeEach(() => {
  atom = {
    config: { get: jest.fn(() => true) },
    tooltips: { add: jest.fn() },
  };
});

describe('createStatusTile()', () => {
  test('it creates a div with "Prettier" and tooltip status indicating formatOnSave status', () => {
    global.document = {
      createElement: jest.fn(() => ({
        classList: {
          add: jest.fn(),
        },
        dataset: {},
        appendChild: jest.fn(),
      })),
      createTextNode: jest.fn(arg => arg),
    };
    
    const div = createStatusTile();
    
    expect(div.dataset.formatOnSave).toBe('enabled');
    expect(div.classList.add).toHaveBeenCalledWith('prettier-atom-status-tile');
    expect(div.appendChild).toHaveBeenCalledWith('Prettier');
  });
});

describe('updateStatusTile()', () => {
  it('disposes previous tooltip and adds a new one with the current formatOnSave status', () => {
    const div = {};
    const disposable = { add: jest.fn() };

    updateStatusTile(disposable, div);

    expect(disposeCurrentTooltip).toHaveBeenCalled(); // assuming you've made this an exported function and mocked it
    expect(atom.tooltips.add).toHaveBeenCalledWith(div, {title: 'Format on Save: enabled'});
  });
});
```